src/main/java/uk/gov/hmcts/divorce/systemupdate/schedule/conditionalorder/SystemResendCOPronouncedCoverLettersTask.java

```diff
@@ -0,0 +1,143 @@
+        final String serviceAuth = authTokenGenerator.generate();
+
+        try {
+            final BoolQueryBuilder query =
+                boolQuery()
+                    .must(matchQuery(STATE, ConditionalOrderPronounced))
+                    .must(
+                        boolQuery()
+                            .should(
+                                boolQuery()
+                                    .must(matchQuery(String.format(DATA, APPLICANT1_OFFLINE), YES))
+                                    .must(matchQuery(String.format(DATA, APPLICANT1_PRIVATE_CONTACT), PRIVATE.getType()))
+                            )
+                            .should(
+                                boolQuery()
+                                    .must(matchQuery(String.format(DATA, APPLICANT2_OFFLINE), YES))
+                                    .must(matchQuery(String.format(DATA, APPLICANT2_PRIVATE_CONTACT), PRIVATE.getType()))
+                            )
+                            .minimumShouldMatch(1)
+                    )
+                    .mustNot(matchQuery(String.format(DATA, NOTIFICATION_FLAG), YES));
+
+            final List<CaseDetails> casesToBeUpdated =
+                ccdSearchService.searchForAllCasesWithQuery(query, user, serviceAuth, ConditionalOrderPronounced);
+
+            for (final CaseDetails caseDetails : casesToBeUpdated) {
+                triggerResendCoPronouncedCoverLetterForEligibleCases(user, serviceAuth, caseDetails);
+            }
+
+            log.info("SystemResendCOPronouncedCoverLettersTask completed.");
+        } catch (final CcdSearchCaseException e) {
+            log.error("SystemResendCOPronouncedCoverLettersTask stopped after search error", e);
+        } catch (final CcdConflictException e) {
+            log.info("SystemResendCOPronouncedCoverLettersTask stopping "
+                + "due to conflict with another running task"
+            );
+        }
+    }
+
+    private void triggerResendCoPronouncedCoverLetterForEligibleCases(User user, String serviceAuth, CaseDetails caseDetails) {
+        try {
+
+            final CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
+
+            if (isCaseEligibleToResendCoverLetter(caseData)) {
+                log.info("Submitting Resend CO Pronounced letter for Case {}", caseDetails.getId());
+                ccdUpdateService.submitEvent(caseDetails, SYSTEM_RESEND_CO_PRONOUNCED_COVER_LETTER, user, serviceAuth);
+            }
+        } catch (final CcdManagementException e) {
+            log.error("Submit event failed for case id: {}, continuing to next case", caseDetails.getId());
+        } catch (final IllegalArgumentException e) {
+            log.error("Deserialization failed for case id: {}, continuing to next case", caseDetails.getId());
+        }
+    }
+
+    private boolean isCaseEligibleToResendCoverLetter(final CaseData caseData) {
```

Comment:
This method is quite complicated is there enough tests to cover all conditions?
Sometimes it may be easier to test if the method is moved into its own component.

Reply:
added few more test cases 

> shouldSubmitEventResendCOPronouncedCoverLetterWhenOfflineApplicant1ContactIsPrivateAndCoverLetterIsNotInConfidentialList

>shouldSubmitEventResendCOPronouncedCoverLetterWhenOfflineApplicant2ContactIsPrivateAndCoverLetterIsNotInConfidentialList

>shouldNotSubmitEventResendCOPronouncedCoverLetterWhenOfflineApplicantsContactIsPublic

>shouldNotSubmitEventResendCOPronouncedCoverLetterWhenOfflineApplicantsContactIsPrivateButCoverLettersAreAlreadyInConfidentialList